ConfigSpace/util.pyx

```diff
@@ -436,3 +436,185 @@ def fix_types(configuration: dict,
+            Same description as return value of get_value_set()
+
+        hp_names: list of strs
+            List of hyperparameter names
+
+        Returns
+        -------
+        list of dicts
+            List of configuration dicts
+
+        '''
+        grid = []
+        import itertools
+        for i, element in enumerate(itertools.product(*value_sets)):
+            config_dict = {}
+            for j, hp_name in enumerate(hp_names):
+                config_dict[hp_name] = element[j]
+            grid.append(config_dict)
+
+        return grid
+
+
+    value_sets = [] # list of tuples: each tuple within is the grid values to be taken on by a Hyperparameter
+    hp_names = []
+
+    # Get HP names and allowed grid values they can take for the HPs at the top level of ConfigSpace tree
+    for hp_name in configuration_space._children['__HPOlib_configuration_space_root__']:
+        value_sets.append(get_value_set(num_steps_dict, hp_name))
+        hp_names.append(hp_name)
+
+    # Create a Cartesian product of above allowed values for the HPs. Hold them in an "unchecked" list because some of the conditionally dependent HPs may become active for some of the elements of the Cartesian product and in these cases creating a Configuration would throw an Error (see below).
+    unchecked_grid_pts = get_cartesian_product(value_sets, hp_names) # Creates a list of Configuration dicts
+    checked_grid_pts = []
+    conditional_grid_lens = []
+
+    while len(unchecked_grid_pts) > 0:
+        try:
+            grid_point = Configuration(configuration_space, unchecked_grid_pts[0])
+            checked_grid_pts.append(grid_point)
+        except ValueError as e:
+            value_sets = []
+            hp_names = []
+            new_active_hp_names = []
+
+            for hp_name in unchecked_grid_pts[0]: # "for" loop over currently active HP names
+                value_sets.append(tuple([unchecked_grid_pts[0][hp_name], ]))
+                hp_names.append(hp_name)
+                for new_hp_name in configuration_space._children[hp_name]: # Checks if the conditionally dependent children of already active HPs are now active
+                    if new_hp_name not in new_active_hp_names and new_hp_name not in unchecked_grid_pts[0]:
+                        all_cond_ = True
+                        for cond in configuration_space._parent_conditions_of[new_hp_name]:
+                            if not cond.evaluate(unchecked_grid_pts[0]):
+                                all_cond_ = False
+                        if all_cond_:
+                            new_active_hp_names.append(new_hp_name)
+
+
+            for hp_name in new_active_hp_names:
+                value_sets.append(get_value_set(num_steps_dict, hp_name))
+                hp_names.append(hp_name)
+            if len(new_active_hp_names) > 0: # this check might not be needed, as there is always going to be a new active HP when in this except block? #TODO

```

Comment:
you could add an else that should never be triggered just to be sure?

Reply:
And throw an Error inside that else?

Reply:
Yes, exactly.

Reply:
Done.